website/server/libs/payments.js

```diff
@@ -128,13 +137,26 @@ api.createSubscription = async function createSubscription (data) {
     }
   }
 
-  await data.user.save();
+  if (group) {
+    await group.save();
+  } else {
+    await data.user.save();
+  }
+
   if (data.gift) await data.gift.member.save();
 };
 
 // Sets their subscription to be cancelled later
 api.cancelSubscription = async function cancelSubscription (data) {
   let plan = data.user.purchased.plan;
+  let group;
```

Comment:
sometimes cancelSubscription is passed groupId and sometimes the entire group https://github.com/HabitRPG/habitica/pull/8153/files#diff-1e4a176d299b36b0817c230e499db62dR287